CRM/Contact/BAO/GroupContactCache.php

```diff
@@ -256,10 +260,12 @@ protected static function flushCaches() {
       return;
     }
     $params = [1 => [self::getCacheInvalidDateTime(), 'String']];
-    $groupsDAO = CRM_Core_DAO::executeQuery("SELECT id FROM civicrm_group WHERE cache_date <= %1", $params);
+    $groupsToFlushSQL = 'SELECT group_id FROM civicrm_group_contact_cache gc INNER JOIN civicrm_group g
+WHERE cache_date <= %1 OR cache_date IS NULL GROUP BY group_id';
+    $groupsDAO = CRM_Core_DAO::executeQuery($groupsToFlushSQL, $params);
     $expiredGroups = [];
     while ($groupsDAO->fetch()) {
-      $expiredGroups[] = $groupsDAO->id;
+      $expiredGroups[] = $groupsDAO->group_id;
```

Comment:
@eileenmcnaughton Alternative based on your suggestion. This only runs the delete query if there is actually data to be cleared out. This also prevents a DELETE being run when cache_date was set but expired and the group had no records (or was already cleared somehow!).

Reply:
@mattwire my suspicion is that inclusing the `civicrm_group_contact_cache` in the query will mean it potentially returns thousands and thousands or results. I think 2 quick queries (1 to get the groups & 1 to check if there is at least one row to delete) will be quicker than 1 slow one